tests/test_initial_data_preparation.py

```diff
@@ -0,0 +1,276 @@
+import pytest
+import pandas as pd
+from utils.initial_data_preparation import (
+    mark_incomplete_urls,
+    get_base_repo_url,
+    extract_and_flag_domains,
+)
+
+
+def test_get_base_repo_url():
+    # Test data setup
+    data = pd.DataFrame(
+        {
+            "repourl": [
+                None,  # Test for None input
+                "https://github.com/getdnsapi/stubby",
+                "https://github.com/namecoin",
+                "https://github.com/osresearch/heads/issues/540",
```

Comment:
Can you add a valid url that starts with http:// please?

Reply:
I have added `http://git.savannah.gnu.org/cgit/gnucap.git` --> The test passes. 
However, I do not expect to see any `repourl` starting with `http` here as `http` is converted to 'https' in the lines before the call for the function `get_base_repo_url` .

Reply:
IIRC there are 15 URLs in the dataset from NLnet that have `http://` e.g. `http://devel.ag-projects.com/repositories/sylkserver`

Reply:
Yes. I have noticed them as well.